### PR TITLE
[Misc] Update instance info of engine in a timely manner

### PR DESCRIPTION
**Pull request by s5u13b**
Previously, we updated the instance info of the engine after each step. Instance info is used by the global scheduler to get the instance's memory usage, etc. However, the actual time point of instance memory update occurs when the scheduler schedules the requests before each step. So, instead of updating the instance info after each step, this pr implements the scheduler to call back the engine to update instance info right after scheduling requests. With this change, the global scheduler can get more timely instance info, which could benefit dynamic scheduling.
Also, I fix some arguments description and TODOs passingly in this pr.

### CODE REVIEW

File: docs/Arguments.md [hunk 1]
```diff
@@ -104,15 +107,15 @@ usage: -m llumnix.entrypoints.vllm.api_server [-h]
...
 
 `--scale-up-threshold`
 - Scale up threshold.
-- Default: 4
+- Default: 10
 
 `--scale-down-threshold`
 - Scale down threshold.
-- Default: 100
+- Default: 60
```
Comment (zhypku) [inline docs/Arguments.md:118]: why change these default values?

File: llumnix/arg_utils.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ class EngineManagerArgs:
     gpu_type: str = "a10"
     migration_backend_init_timeout: float = 10.0
     migration_backend: str = "rpc"
-    migration_cache_blocks: int = 32
+    migration_cache_blocks: int = 512
```
Comment (zhypku) [inline llumnix/arg_utils.py:57]: and here

File: llumnix/backends/vllm/scheduler.py [hunk 3]
```diff
@@ -185,8 +190,7 @@ def free_src_request(self, backend_request: SequenceGroup) -> None:
         logger.info("free seq {}".format(seq.seq_id))
         self.free_seq(seq)
 
-    @scheduler_lock
```
Comment (zhypku) [inline llumnix/backends/vllm/scheduler.py:188]: you dont need lock now?

File: docs/Arguments.md (hunk 1 above)
Comment (s5u13b) [reply docs/Arguments.md:118]: The default values in Arguments.md and arg_utils.py is not consistent. So I change the default values in Arguments.md to be consistent with the values in arg_utils.py.

Comment (zhypku) [reply docs/Arguments.md:118]: I see

File: llumnix/backends/vllm/scheduler.py (hunk 3 above)
Comment (s5u13b) [reply llumnix/backends/vllm/scheduler.py:188]: Yes. Previously, get_instance_info is called by engine. Now, _get_instance_info is only called by schedule function in scheduler. And schedule function already have lock, so there is no need to lock _get_instance_info.

**Review by zhypku — APPROVED** [review]

**Review by KuilongCui — APPROVED** [review]

**Pull request merged**